k8s/upgrades/0.8.2-0.9.0/cstor/cstor-pool-update-082-090.yaml

```diff
@@ -0,0 +1,664 @@
+
+    {{- $taskStatus := upgradeResultWithTaskStatus $status -}}
+    {{- $taskMessage := upgradeResultWithTaskMessage $message -}}
+    {{- $taskName := upgradeResultWithTaskName "upgrade-cstor-pool-082-090-get-cstorpool" -}}
+    {{- $URName := upgradeResultWithTaskOwnerName .UpgradeItem.upgradeResultName -}}
+    {{- $URNamespace := upgradeResultWithTaskOwnerNamespace "default" -}}
+    {{- $taskStartTime := upgradeResultWithTaskStartTime now -}}
+    {{- $taskEndTime := upgradeResultWithTaskEndTime now -}}
+    {{- upgradeResultUpdateTasks $taskStartTime $URName $URNamespace $taskName $taskStatus $taskMessage $taskEndTime -}}
+    {{- if eq $status .Config.failStatus.value }}
+    {{- verifyErr "CStorPool version is not 0.8.2" true | saveAs "CStorPool.verifyErr" .TaskResult | noop -}} 
+    {{- end }}
+---
+# upgrade-cstor-pool-082-090-get-cstorpool gets details of Pool Deployment.
+# This runtask is for get object info which will use in some runtask.
+# This should always run.
+apiVersion: openebs.io/v1alpha1
+kind: RunTask
+metadata:
+  name: upgrade-cstor-pool-082-090-get-deployment
+  namespace: default
+spec:
+  meta: |
+    id: poolDeployment
+    apiVersion: extensions/v1beta1
+    kind: Deployment
+    action: get
+    objectName: {{ .UpgradeItem.name }}
+    runNamespace: {{ .UpgradeItem.namespace }}
+    disable: false
+  post: |
+    {{- jsonpath .JsonResult "{.spec.template.spec.containers[?(@.name=='cstor-pool')].image}" | trim | saveAs "poolDeployment.cstorpoolimage" .TaskResult | noop -}}
+    {{- jsonpath .JsonResult "{.spec.template.spec.containers[?(@.name=='cstor-pool-mgmt')].image}" | trim | saveAs "poolDeployment.cstorpoolmgmtimage" .TaskResult | noop -}}
+    {{- jsonpath .JsonResult "{.metadata.labels.openebs\\.io/version}" | trim | saveAs "poolDeployment.labels" .TaskResult | noop -}}
+
+    {{- $message := printf "Successfully got details of CStorPool deployment {%s}." .UpgradeItem.name -}}
+    {{- $status :="" }}
+    {{- if eq .TaskResult.poolDeployment.labels "0.8.2" }}
+    {{- $status =.Config.successStatus.value }}
+    {{- else }}
+    {{- $status =.Config.failStatus.value }}
+    {{- end }}
+
+    {{- $taskStatus := upgradeResultWithTaskStatus $status -}}
+    {{- $taskMessage := upgradeResultWithTaskMessage $message -}}
+    {{- $taskName := upgradeResultWithTaskName "upgrade-cstor-pool-082-090-get-deployment" -}}
+    {{- $URName := upgradeResultWithTaskOwnerName .UpgradeItem.upgradeResultName -}}
+    {{- $URNamespace := upgradeResultWithTaskOwnerNamespace "default" -}}
+    {{- $taskStartTime := upgradeResultWithTaskStartTime now -}}
+    {{- $taskEndTime := upgradeResultWithTaskEndTime now -}}
+    {{- upgradeResultUpdateTasks $taskStartTime $URName $URNamespace $taskName $taskStatus $taskMessage $taskEndTime -}}
+    {{- if eq $status .Config.failStatus.value }}
+    {{- verifyErr "CStorPool deployment version is not 0.8.2" true | saveAs "poolDeployment.verifyErr" .TaskResult | noop -}} 
```

Comment:
check my earlier comments on grouping all messages together & here just use the variable e.g. `$errMsgInvalidVersion`
```yaml
{{- verifyErr $errMsgInvalidVersion true | saveAs "poolDeployment.verifyErr" .TaskResult | noop -}}
```